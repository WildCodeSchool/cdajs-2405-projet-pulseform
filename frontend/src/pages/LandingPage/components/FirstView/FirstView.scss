@use "@assets/variables" as *;

.first-view {
  height: 100vh;

  &__blob {
    position: absolute;
    right: 0;
    width: 268px;
    top: 145px;
    overflow: hidden;
    z-index: 0;

    @media screen and (min-width: 500px) {
      width: 42vw;
      height: 74vh;
      top: 16vh;
      right: 14vw;
      overflow: unset;
    }

    &__img {
      width: 440px;
      height: 347px;

      @media screen and (min-width: 500px) {
        width: 100%;
        height: 100%;
      }
    }
  }

  &__girl {
    position: absolute;
    right: 0;
    top: 300px;
    width: 268px;
    z-index: 0;
    overflow: hidden;

    @media screen and (min-width: 500px) {
      width: 60vw;
      height: 80vh;
      top: 138px;
      right: 40px;
      overflow: unset;
    }

    &__img {
      width: 400px;
      height: 500px;

      @media screen and (min-width: 500px) {
        width: 100%;
        height: 100%;
      }
    }
  }

  &__header {
    display: flex;
    align-items: center;
    justify-content: space-around;
    padding: 32px 0 40px;

    @media screen and (min-width: 500px) {
      justify-content: space-between;
      margin: 0 96px;
    }

    &__logo-m {
      display: block;
      @media screen and (min-width: 500px) {
        display: none;
      }
    }

    &__logo-d {
      display: none;
      @media screen and (min-width: 501px) {
        display: block;
      }
    }

    &__container {
      display: flex;
      gap: 16px;
      align-items: center;

      &__buttom {
        display: none;
        @media screen and (min-width: 500px) {
          display: block;

          &__width {
            width: 200px;
          }
        }
      }

      &__connect {
        @include h2-phone;
        color: $white;
        text-decoration: none;
        &:hover {
          cursor: pointer;
        }
      }
    }
  }

  &__description {
    @include landing-page-title-phone;
    position: relative;
    z-index: 1;
    color: $primary;
    width: 180px;
    text-align: center;
    margin-left: 7%;

    @media screen and (min-width: 500px) {
      @include landing-page-title-desktop;
      width: 486px;
      margin-left: 10%;
    }
  }

  &__buttom {
    position: absolute;
    display: flex;
    justify-content: center;
    z-index: 1;
    bottom: 4%;
    width: 100%;

    @media screen and (min-width: 500px) {
      display: none;
    }
  }

  &__buttom-program {
    display: none;

    @media screen and (min-width: 501px) {
      @include h1-phone;
      display: flex;
      justify-content: center;
      z-index: 1;
      position: absolute;
      bottom: 10%;
      width: 70%;

      &__button {
        height: 80px;
        font-size: 24px;
      }
    }
  }
}
