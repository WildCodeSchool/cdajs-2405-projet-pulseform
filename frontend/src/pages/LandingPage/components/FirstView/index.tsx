import BasicButton from "@components/atoms/BasicButton";
import LittleLogo from "@components/atoms/LittleLogo";

import blobFirstView from "@assets/blobs/blob.svg";
import girldImage from "@assets/images/girl.svg";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
import "./FirstView.scss";

function FirstView() {
	const { t } = useTranslation();

	return (
		<div className="first-view">
			<div className="first-view__blob">
				<img className="first-view__blob__img" src={blobFirstView} alt="blob" />
			</div>
			<div className="first-view__girl">
				<img className="first-view__girl__img" src={girldImage} alt="girl" />
			</div>
			<div className="first-view__header">
				<div className="first-view__header__logo-m">
					<LittleLogo hasLabel size="mobile" />
				</div>
				<div className="first-view__header__logo-d">
					<LittleLogo hasLabel />
				</div>
				<div className="first-view__header__container">
					<div className="first-view__header__container__buttom">
						<BasicButton className="first-view__header__container__buttom__width">
							{t("CREATE_ACCOUNT")}
						</BasicButton>
					</div>
					<Link to="/login" className="first-view__header__container__connect">
						{t("CONNECT_MY")}
					</Link>
				</div>
			</div>
			<p className="first-view__description">{t("LANDING_DESCRIPTION")}</p>
			<div className="first-view__buttom">
				<BasicButton>{t("CREATE_ACCOUNT")}</BasicButton>
			</div>
			<div className="first-view__buttom-program">
				<BasicButton className="first-view__buttom-program__button">
					{t("TEST_PROGRAM")}
				</BasicButton>
			</div>
		</div>
	);
}

export default FirstView;
