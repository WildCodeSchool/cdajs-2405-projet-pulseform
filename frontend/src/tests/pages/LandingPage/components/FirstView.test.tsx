import { MockedProvider } from "@apollo/client/testing";
import FirstView from "@pages/LandingPage/components/FirstView";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("FirstView", () => {
  it("renders the FirstView component", () => {
    const tree = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <MemoryRouter>
          <FirstView />
        </MemoryRouter>
      </MockedProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
