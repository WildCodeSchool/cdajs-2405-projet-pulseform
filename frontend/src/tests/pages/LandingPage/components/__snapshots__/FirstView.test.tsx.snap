// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstView > renders the FirstView component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="first-view"
      >
        <div
          class="first-view__blob"
        >
          <img
            alt="blob"
            class="first-view__blob__img"
            src="/src/assets/blobs/blob.svg"
          />
        </div>
        <div
          class="first-view__girl"
        >
          <img
            alt="girl"
            class="first-view__girl__img"
            src="/src/assets/images/girl.svg"
          />
        </div>
        <div
          class="first-view__header"
        >
          <div
            class="first-view__header__logo-m"
          >
            <div
              class="little-logo little-logo--mobile"
            >
              <img
                alt="Logo Pulse Form"
                class="little-logo__img"
                src="/src/assets/icons/mobile-logo.svg"
              />
              <p
                class="little-logo__title little-logo__title--mobile"
              >
                Pulse Form
              </p>
            </div>
          </div>
          <div
            class="first-view__header__logo-d"
          >
            <div
              class="little-logo little-logo--desktop"
            >
              <img
                alt="Logo Pulse Form"
                class="little-logo__img"
                src="/src/assets/icons/mobile-logo.svg"
              />
              <p
                class="little-logo__title little-logo__title--desktop"
              >
                Pulse Form
              </p>
            </div>
          </div>
          <div
            class="first-view__header__container"
          >
            <div
              class="first-view__header__container__buttom"
            >
              <button
                class="basic-button first-view__header__container__buttom__width basic-button--basic"
                type="button"
              >
                CREATE_ACCOUNT
              </button>
            </div>
            <a
              class="first-view__header__container__connect"
              href="/login"
            >
              CONNECT_MY
            </a>
          </div>
        </div>
        <p
          class="first-view__description"
        >
          LANDING_DESCRIPTION
        </p>
        <div
          class="first-view__buttom"
        >
          <button
            class="basic-button basic-button--basic"
            type="button"
          >
            CREATE_ACCOUNT
          </button>
        </div>
        <div
          class="first-view__buttom-program"
        >
          <button
            class="basic-button first-view__buttom-program__button basic-button--basic"
            type="button"
          >
            TEST_PROGRAM
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="first-view"
    >
      <div
        class="first-view__blob"
      >
        <img
          alt="blob"
          class="first-view__blob__img"
          src="/src/assets/blobs/blob.svg"
        />
      </div>
      <div
        class="first-view__girl"
      >
        <img
          alt="girl"
          class="first-view__girl__img"
          src="/src/assets/images/girl.svg"
        />
      </div>
      <div
        class="first-view__header"
      >
        <div
          class="first-view__header__logo-m"
        >
          <div
            class="little-logo little-logo--mobile"
          >
            <img
              alt="Logo Pulse Form"
              class="little-logo__img"
              src="/src/assets/icons/mobile-logo.svg"
            />
            <p
              class="little-logo__title little-logo__title--mobile"
            >
              Pulse Form
            </p>
          </div>
        </div>
        <div
          class="first-view__header__logo-d"
        >
          <div
            class="little-logo little-logo--desktop"
          >
            <img
              alt="Logo Pulse Form"
              class="little-logo__img"
              src="/src/assets/icons/mobile-logo.svg"
            />
            <p
              class="little-logo__title little-logo__title--desktop"
            >
              Pulse Form
            </p>
          </div>
        </div>
        <div
          class="first-view__header__container"
        >
          <div
            class="first-view__header__container__buttom"
          >
            <button
              class="basic-button first-view__header__container__buttom__width basic-button--basic"
              type="button"
            >
              CREATE_ACCOUNT
            </button>
          </div>
          <a
            class="first-view__header__container__connect"
            href="/login"
          >
            CONNECT_MY
          </a>
        </div>
      </div>
      <p
        class="first-view__description"
      >
        LANDING_DESCRIPTION
      </p>
      <div
        class="first-view__buttom"
      >
        <button
          class="basic-button basic-button--basic"
          type="button"
        >
          CREATE_ACCOUNT
        </button>
      </div>
      <div
        class="first-view__buttom-program"
      >
        <button
          class="basic-button first-view__buttom-program__button basic-button--basic"
          type="button"
        >
          TEST_PROGRAM
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
